src/test/groovy/org/sonarqube/gradle/FunctionalTests.groovy

```diff
@@ -344,15 +344,138 @@ class FunctionalTests extends Specification {
+                .build()
 
 
         then:
         noExceptionThrown()
     }
+
+    def "scan all is enabled"() {
+        given:
+        settingsFile << "rootProject.name = 'java-task-toolchains'"
+        buildFile << """
+        plugins {
+            id 'java'
+            id 'org.sonarqube'
+        }
+        """
+
+        when:
+        def result = GradleRunner.create()
+                .withGradleVersion(gradleVersion)
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments('sonar', '--info',
+                        '-Dsonar.gradle.scanAll=true',
+                        '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath())
+                .withPluginClasspath()
+                .build()
+
+        then:
+        result.task(":sonar").outcome == SUCCESS
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+        props."sonar.gradle.scanAll" == "true"
+        result.output.contains("Parameter sonar.gradle.scanAll is enabled. The scanner will attempt to collect additional sources.")
+
+        var sources = ((String) props."sonar.sources").split(",")
+        sources.size() == 3
+        sources[0].endsWith("""$testProjectDir/build.gradle""")
+        sources[1].endsWith("""$testProjectDir/gradle.properties""")
+        sources[2].endsWith("""$testProjectDir/settings.gradle""")
+    }
+
+    def "scan all is enabled but not applied"() {
+        given:
+        settingsFile << "rootProject.name = 'java-task-toolchains'"
+        buildFile << """
+        plugins {
+            id 'java'
+            id 'org.sonarqube'
+        }
+        """
+
+        when:
+        var arguments = ['sonar', '--info',
+                         '-Dsonar.gradle.scanAll=true',
+                         sonarSourcesOverride != null ? '-Dsonar.sources=' + sonarSourcesOverride : null,
+                         sonarTestsOverride != null ? '-Dsonar.tests=' + sonarTestsOverride : null,
+                         '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath()]
+        def result = GradleRunner.create()
+                .withGradleVersion(gradleVersion)
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments(arguments.stream().filter { it != null }.toList())
+                .withPluginClasspath()
+                .build()
+
+        then:
+        result.task(":sonar").outcome == SUCCESS
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+        props."sonar.gradle.scanAll" == "true"
+        result.output.contains("Parameter sonar.gradle.scanAll is enabled. The scanner will attempt to collect additional sources.")
+        result.output.contains("Parameter sonar.gradle.scanAll is enabled but the scanner will not collect additional sources because sonar.sources or sonar.tests are already set.")
```

Comment:
let's change the end of the message to be more consistent with the Maven implementation by mentioning overriding
```suggestion
        result.output.contains("Parameter sonar.gradle.scanAll is enabled but the scanner will not collect additional sources because sonar.sources or sonar.tests has been overridden..")
```
```suggestion
        result.output.contains("Parameter sonar.gradle.scanAll is enabled but the scanner will not collect additional sources because sonar.sources or sonar.tests are already set.")
```